jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -597,4 +667,100 @@ public String toString()
                 getPooled());
         }
     }
+
+    public interface Holder<P>
+    {
+        Pool.Entry<P> getEntry();
+
+        void hold();
+
+        void free();
```

Comment:
How about renaming this method `weaken()`?

Reply:
I prefer `acquire()` as it matches where it is called from.

Reply:
@sbordet again it is not an acquire.  perhaps `firstAcquire`?     I'll go with `weaken` for now unless there is consensus otherwise.